src/image.c

```diff
@@ -230,3 +230,183 @@ wdGetImageSize(WD_HIMAGE hImage, UINT* puWidth, UINT* puHeight)
     }
 }
 
+static void 
+wdBufferRGB2Bitmap(BYTE* Scan0, INT Stride, UINT channels, UINT width, UINT height, const BYTE *rgb)
+{
+  UINT i, j;
+
+  for (j = 0; j < height; j++)
+  {
+    UINT line_offset = j * width * 3;
+    const BYTE* rgb_line = rgb + line_offset;
+    BYTE* line_data = Scan0 + j*Stride;
+
+    for (i = 0; i < width; i++)
+    {
+      int offset = i * 3;
+      int offset_data = i * channels;
+      line_data[offset_data + 0] = rgb_line[offset + 2];  /* Blue */
+      line_data[offset_data + 1] = rgb_line[offset + 1];  /* Green */
+      line_data[offset_data + 2] = rgb_line[offset + 0];  /* Red */
+
+      if (channels == 4)
+        line_data[offset_data + 3] = 255;
+    }
+  }
+}
+
+static void 
+wdBufferRGBA2Bitmap(BYTE* Scan0, INT Stride, BOOL PreAlpha, UINT width, UINT height, const BYTE* rgba)
+{
+  UINT i, j;
+
+  for (j = 0; j < height; j++)
+  {
+    UINT line_offset = j * width * 4;
+    const BYTE* rgba_line = rgba + line_offset;
+    BYTE* line_data = Scan0 + j*Stride;
+
+    for (i = 0; i < width; i++)
+    {
+      int offset = i * 4;
+
+      if (PreAlpha)
+      {
+        line_data[offset + 0] = (rgba_line[offset + 2] * rgba_line[offset + 3]) / 255;  /* Blue */
+        line_data[offset + 1] = (rgba_line[offset + 1] * rgba_line[offset + 3]) / 255;  /* Green */
+        line_data[offset + 2] = (rgba_line[offset + 0] * rgba_line[offset + 3]) / 255;  /* Red */
```

Comment:
Have you tested this code path? I think the multiplication of the two bytes can overflow and we then get garbage.
I would add some explicit casting:
``` C
line_data[offset + 0] = ((int)rgba_line[offset + 2] * (int)rgba_line[offset + 3]) / 255;
```

Reply:
Yes, I tested this. It works without the type cast.